core/appeal/service.go

```diff
@@ -534,6 +534,98 @@ func (s *Service) Revoke(ctx context.Context, id string, actor, reason string) (
 	return revokedAppeal, nil
 }
 
+func (s *Service) BulkRevoke(ctx context.Context, filters *domain.ListAppealsFilter, actor, reason string) ([]*domain.Appeal, error) {
+	result := make([]*domain.Appeal, 0)
+	filters.Statuses = []string{domain.AppealStatusActive}
+	appeals, err := s.Find(ctx, filters)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(appeals) == 0 {
+		return nil, nil
+	}
+
+	batchSize := 10
+	timeLimiter := make(chan int, batchSize)
+
+	for i := 1; i <= batchSize; i++ {
+		timeLimiter <- i
+	}
+
+	go func() {
+		for range time.Tick(1 * time.Second) {
+			for i := 1; i <= batchSize; i++ {
+				timeLimiter <- i
+			}
+		}
+	}()
+
+	totalRequests := len(appeals)
+	done := make(chan *domain.Appeal, totalRequests)
+	resourceAppealMap := make(map[string][]*domain.Appeal, 0)
+
+	for _, appeal := range appeals {
+		var resourceAppeals []*domain.Appeal
+		var ok bool
+		if resourceAppeals, ok = resourceAppealMap[appeal.ResourceID]; ok {
+			resourceAppeals = append(resourceAppeals, appeal)
+		} else {
+			resourceAppeals = []*domain.Appeal{appeal}
+		}
+		resourceAppealMap[appeal.ResourceID] = resourceAppeals
+	}
+
+	for _, resourceAppeals := range resourceAppealMap {
+		go s.expiredInActiveUserAppeal(ctx, timeLimiter, done, actor, reason, resourceAppeals)
```

Comment:
Any reason why we need to parallelized here? How is the `totalRequests` that we expected?
For each go routine there is an update task to DB. So this will create mutiple concurrent row updates at the same time.

Reply:
this can be 10-50 based on the frequency of job runs.